README.md

```diff
@@ -104,6 +104,20 @@ item 0: User {
 }
 ```
 
+#### Matching with general function mapping
+
+You can also apply general functions/lambdas for matches:
+```java
+assertThat("A string", MapMatcher.map(s -> s.charAt(0), equalTo('A')))
```

Comment:
I'm curious if you could describe the intended uses a little bit more or provide some more examples - couldn't this example also be solved with `assertThat("A string", startsWith("a"))`? 

http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#startsWith(java.lang.String)

Just curious what other uses you have in mind

Reply:
The real usecase that made me look for something like this is:
```java
assertThat(protobufObj, map(UserAttributesMutated::getAttributesList, hasItem(map(AttributeMutation::getName, equalTo("country")))))));
```

Reply:
I would say that this use-case is already fulfilled by the POJO matcher.

    assertThat(protobufObj, is(pojo(UserAttributesMutated.class).where(UserAttributesMutated::getAttributesList, hasItem(...)))

We can probably make it more concise but that should be a PR for the POJO matcher.

Also, `MapMatcher` to me sounds like something matching on `Map` objects, and using the word `map` goes a bit against the declarative wording of hamcrest.  I would in any case prefer the wording `resultOf` or something.